src/tests/ftest/util/server_utils.py

```diff
@@ -622,3 +626,165 @@ def get_environment_value(self, name):
+                raise error
+            checks += 1
+            time.sleep(1)
+        if daos_state not in valid_states:
+            raise ServerFailed(
+                "Error checking DAOS state, currently neither {} after "
+                "{} state check(s)!".format(valid_states, checks))
+        return daos_state
+
+    def system_start(self):
+        """Start the DAOS IO servers.
+
+        Raises:
+            ServerFailed: if there was an error starting the servers
+
+        """
+        self.log.info("Starting DAOS IO servers")
+        self.check_system_state(("stopped"))
+        self.dmg.system_start()
+        if self.dmg.result.exit_status != 0:
+            raise ServerFailed(
+                "Error starting DAOS:\n{}".format(self.dmg.result))
+
+    def system_stop(self):
+        """Stop the DAOS IO servers.
+
+        Raises:
+            ServerFailed: if there was an error stopping the servers
+
+        """
+        self.log.info("Stopping DAOS IO servers")
+        self.check_system_state(("started", "joined"))
+        self.dmg.system_stop()
+        if self.dmg.result.exit_status != 0:
+            raise ServerFailed(
+                "Error stopping DAOS:\n{}".format(self.dmg.result))
+
+    def get_available_storage(self):
+        """Get the available SCM and NVMe storage.
+
+        Raises:
+            ServerFailed: if there was an error stopping the servers
+
+        Returns:
+            list: a list of Bytes objects representing the maximum available
+                SCM size and NVMe size
+
+        """
+        using_dcpm = self.manager.job.using_dcpm
+        using_nvme = self.manager.job.using_nvme
+
+        if using_dcpm or using_nvme:
+            # Stop the DAOS IO servers in order to be able to scan the storage
+            self.system_stop()
+
+            # Scan all of the hosts for their SCM and NVMe storage
+            self.dmg.hostlist = self._hosts
+            result = self.dmg.storage_scan(verbose=True)
+            self.dmg.hostlist = self.get_config_value("access_points")
+            if self.dmg.result.exit_status != 0:
+                raise ServerFailed(
+                    "Error obtaining DAOS storage:\n{}".format(result))
+
+            # Find the sizes of the SCM and NVMe storage configured for use by
+            # the DAOS IO servers.  Convert the lists into hashable tuples so
+            # they can be used as dictionary keys.
+            scm_keys = [
+                os.path.basename(path)
+                for path in self.get_config_value("scm_list")
+                if path
+            ]
+            nvme_keys = self.get_config_value("bdev_list")
+            device_capacities = {}
+            for key in scm_keys + nvme_keys:
+                device_capacities[key] = None
+            for key in device_capacities:
+                regex = r"(?:{})\s+.*\d+\s+([0-9\.]+)\s+([A-Z])B".format(key)
+                data = re.findall(regex, result.stdout)
```

Comment:
(pylint-no-member) Instance of 'dict' has no 'stdout' member

Reply:
Fixed